public/app/plugins/panel/alertlist/module.tsx

```diff
@@ -140,3 +142,136 @@ export const plugin = new PanelPlugin<AlertListOptions>(AlertList)
+            { label: 'Time (desc)', value: SortOrder.TimeDesc },
+          ],
+        },
+        defaultValue: SortOrder.AlphaAsc,
+        category: ['Options'],
+      })
+      .addBooleanSwitch({
+        path: 'dashboardAlerts',
+        name: 'Alerts from this dashboard',
+        defaultValue: false,
+        category: ['Options'],
+      })
+      .addTextInput({
+        path: 'alertName',
+        name: 'Alert name',
+        defaultValue: '',
+        category: ['Filter'],
+        showIf: showIfCurrentState,
+      })
+      .addTextInput({
+        path: 'dashboardTitle',
+        name: 'Dashboard title',
+        defaultValue: '',
+        category: ['Filter'],
+        showIf: showIfCurrentState,
+      })
+      .addCustomEditor({
+        path: 'folderId',
+        name: 'Folder',
+        id: 'folderId',
+        defaultValue: null,
+        editor: function RenderFolderPicker(props) {
+          return (
+            <FolderPicker
+              initialFolderId={props.value}
+              initialTitle="All"
+              enableReset={true}
+              onChange={({ id }) => props.onChange(id)}
+            />
+          );
+        },
+        category: ['Filter'],
+        showIf: showIfCurrentState,
+      })
+      .addCustomEditor({
+        id: 'tags',
+        path: 'tags',
+        name: 'Tags',
+        description: '',
+        defaultValue: [],
+        editor(props) {
+          return <TagsInput tags={props.value} onChange={props.onChange} />;
+        },
+        category: ['Filter'],
+        showIf: showIfCurrentState,
+      })
+      .addBooleanSwitch({
+        path: 'stateFilter.ok',
+        name: 'Ok',
+        defaultValue: false,
+        category: ['State filter'],
+        showIf: showIfCurrentState,
+      })
+      .addBooleanSwitch({
+        path: 'stateFilter.paused',
+        name: 'Paused',
+        defaultValue: false,
+        category: ['State filter'],
+        showIf: showIfCurrentState,
+      })
+      .addBooleanSwitch({
+        path: 'stateFilter.no_data',
+        name: 'No data',
+        defaultValue: false,
+        category: ['State filter'],
+        showIf: showIfCurrentState,
+      })
+      .addBooleanSwitch({
+        path: 'stateFilter.execution_error',
+        name: 'Execution error',
+        defaultValue: false,
+        category: ['State filter'],
+        showIf: showIfCurrentState,
+      })
+      .addBooleanSwitch({
+        path: 'stateFilter.alerting',
+        name: 'Alerting',
+        defaultValue: false,
+        category: ['State filter'],
+        showIf: showIfCurrentState,
+      })
+      .addBooleanSwitch({
+        path: 'stateFilter.pending',
+        name: 'Pending',
+        defaultValue: false,
+        category: ['State filter'],
+        showIf: showIfCurrentState,
+      });
+  })
+  .setMigrationHandler(alertListPanelMigrationHandler);
+
+export const plugin = config.featureToggles.ngalert ? unifiedAlertList : alertList;
```

Comment:
The new alert list is going to have some different options and a separate migration handler so I'm exporting the component based on the feature toggle.